packages/dds/quorum/src/quorum.ts

```diff
@@ -0,0 +1,373 @@
+        if (!proposalValid) {
+            // Drop invalid proposals on the ground.  If delete() returns a promise we will need to resolve it though.
+            return;
+        }
+
+        const accepted = currentValue?.accepted;
+
+        this.values.set(key, { accepted, pending: { type: "delete", sequenceNumber: deleteSequenceNumber }});
+
+        this.emit("pending", key);
+
+        // Emit a noop upon a new key entering pending state, which is how we'll eventually advance it to accepted
+        // TODO: Doesn't work if there's a holdout client that disconnects prior to sending noop, since we aren't
+        // guaranteed to get any further message advancing the MSN.  Observing client disconnects could work.
+        const noOp: IQuorumNoOpOperation = { type: "noop" };
+        this.submitLocalMessage(noOp);
+    };
+
+    private readonly handleIncomingNoOp = (minimumSequenceNumber: number) => {
+        // Run through each of the values, find any pending that should now be considered settled because the MSN has
+        // passed when they were sequenced.
+        for (const [key, value] of this.values) {
+            const pending = value.pending;
+            if (pending !== undefined && minimumSequenceNumber >= pending.sequenceNumber) {
+                // The pending value has settled
+                if (pending.type === "set") {
+                    this.values.set(key, {
+                        accepted: { value: pending.value, sequenceNumber: minimumSequenceNumber },
+                        pending: undefined,
+                    });
+                } else if (pending.type === "delete") {
+                    this.values.delete(key);
+                }
+                this.emit("accepted", key);
+            }
+        }
+    };
+
+    /**
+     * Create a summary for the quorum
+     *
+     * @returns the summary of the current state of the quorum
+     * @internal
+     */
+    protected summarizeCore(serializer: IFluidSerializer): ISummaryTreeWithStats {
+        const content = [...this.values.entries()];
+        return createSingleBlobSummary(snapshotFileName, JSON.stringify(content));
+    }
+
+    /**
+     * {@inheritDoc @fluidframework/shared-object-base#SharedObject.loadCore}
+     * @internal
+     */
+    protected async loadCore(storage: IChannelStorageService): Promise<void> {
+        const content = await readAndParse<[string, QuorumValue][]>(storage, snapshotFileName);
+        content.forEach(([key, value]) => {
+            this.values.set(key, value);
+        });
+    }
+
+    /**
+     * @internal
+     */
+    protected initializeLocalCore() { }
+
+    /**
+     * @internal
+     */
+    protected onDisconnect() {
+        this.disconnectWatcher.emit("disconnect");
+    }
+
+    /**
+     * Override resubmit core to avoid resubmission on reconnect.
+     * TODO It's probably ok to resubmit these?
```

Comment:
Yes, I believe default (not empty override) implementation is good.
That's the weak spot of existing Quorum - proposal fail on disconnect.

Reply:
Yeah, I think resubmitting should be safe/good.  I plan to experiment with this in the next iteration (along with other state transition logic).